firefox_puppeteer/ui/deck.py

```diff
@@ -0,0 +1,47 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from marionette_driver import By
+
+from firefox_puppeteer.ui_base_lib import UIBaseLib
+
+
+class BaseDeck(UIBaseLib):
+
+    def _create_panel_for_id(self, panel_id):
+        """Creates an instance of :class:`Panel` for the specified panel id.
+
+        :param panel_id: The ID of the panel to create an instance of.
+
+        :returns: :class:`Panel` instance
+        """
+        mapping = {'apply': ApplyPanel,
+                   'applyBillboard': ApplyBillboardPanel,
+                   'checkForUpdates': CheckForUpdatesPanel,
+                   'checkingForUpdates': CheckingForUpdatesPanel,
+                   'downloadAndInstall': DownloadAndInstallPanel,
+                   'downloadFailed': DownloadFailedPanel,
+                   'downloading': DownloadingPanel,
+                   'feedPanel': FeedPanel,
+                   'generalPanel': GeneralPanel,
+                   'mediaPanel': MediaPanel,
+                   'noUpdatesFound': NoUpdatesFoundPanel,
+                   'permPanel': PermissionsPanel,
+                   'securityPanel': SecurityPanel
```

Comment:
@whimboo, I have still not found a way to extract the _create_panel_for_id method without also pulling out all the panel classes. It seems the only things remaining in Deck that are common to both about_window and pageinfo, that can be extracted without also pulling out anything specific to one of these two, are three properties: panels, selected_index, and selected_panel. (These three are also in update_wizard, inside the Wizard class.) Is this what you would like refactored out?